src/app/loginComponents/accounts/external/accounts.component.html

```diff
@@ -83,7 +83,7 @@
                 <strong>Username: {{ row.source | getTokenUsername: tokens }} </strong>
                 <span *ngIf="row.name === 'ORCID'">
                   <img src="../../assets/images/account-logos/{{ row.logo }}" alt="ORCID iD logo" class="orcid-id-logo site-icons-small" />
-                  <a href="https://orcid.org/{{ orcidId$ | async }}"> {{ orcidId$ | async }}</a>
+                  <a href="{{ this.accountsService.getRoot(Dockstore.ORCID_AUTH_URL) }}/{{ orcidId$ | async }}"> {{ orcidId$ | async }}</a>
```

Comment:
I should have noticed this when we spoke yesterday... You generally don't want to call functions like this from Angular templates, because they'll keep re-executing (because Angular doesn't know if it's changed or not). In the component, you should create a property that calls your service method, and then reference the property here.

Reply:
The official "Dockstore Frontend Onboarding" guide says:

"No complicated functions or logic in templates or else it's called a bajillion times"

And, irregardless of whether or not the PR runs afoul of that, the ORCID root URL is static and can be precomputed.  Waste not, want not.  Changed!

Reply:
hmm...maybe i should remove "complicated" from the wiki 🤔

Reply:
I like the current guideline.  A function in a template could be useful sometimes, and as long as the repeated invocations have negligible computational overhead, it shouldn't be a problem.  My understanding (yet to be confirmed by looking at the generated code) is that the Angular change detection logic will evaluate each chunk of embedded code but will only re-render if the result changes.

Reply:
Quick benchmark: the getRoot() replace() invocation runs at about 2,000,000 ~24-char-long URLs per second on my Mac, node v14.17.0.

Reply:
Will the requested change affect the user perception of performance in any noticeable way? Almost certainly not.

It's just a best practice. If 20 function calls  crept into the page, then it might start to be noticeable. And it's a learning PR for you, so when you write more complex pages, you should leverage the techniques you learn here.

A poor Java analogy: if it's called only once in an app, one could use StringBuffer instead of StringBuilder, and there would be no noticeable difference in performance. But if I saw a PR, I would still recommend it be changed -- maybe in the future the code gets copied as a model, invoked from a loop, etc.

Reply:
Yes, Charles, your recommendation was good, thanks!